Sources/SpeziDevicesUI/Devices/DevicesTab.swift

```diff
@@ -0,0 +1,82 @@
+//
+// This source file is part of the Stanford Spezi open-project
+//
+// SPDX-FileCopyrightText: 2024 Stanford University
+//
+// SPDX-License-Identifier: MIT
+//
+
+import SpeziBluetooth
+import SpeziDevices
+import SwiftUI
+
+
+/// Devices tab showing grid of paired devices and functionality to pair new devices.
+///
+/// - Note: Make sure to place this view into an `NavigationStack`.
+public struct DevicesTab<PairingHint: View>: View {
```

Comment:
As noted in the ENGAGE-HF review, we might not want to bind the view to only be applicable for a tab. Could also be in a nested menu structure in, e.g., the accounts settings. We should at least be that flexible and reflect that in the name.

Reply:
The idea was, that the `DevicesGrid` already serves as a reusable component and therefore `DeviceTab` exists to have a zero-conf UI component that, e.g., also already packages the NavigationStack (was removed as it is never a good idea to include the NavigationStack in your reusable views if it can be avoided).

I restructured the view to be more flexible and reusable:
* Renamed `DevicesTab` to `DevicesView`.
* Moved toolbar items and navigation title from `DevicesGrid` to `DeviceView`. `DevicesGrid` is now really only the Grid Layout (and placeholder views if there aren't any paired devices) while `DeviceView` provides additional helpful infrastructure like Add Button in the toolbar, search indicators, the sheet for device setup and automatic Bluetooth scanning.